diff --git a/node_modules/react-native-worklets/android/CMakeLists.txt b/node_modules/react-native-worklets/android/CMakeLists.txt
--- a/node_modules/react-native-worklets/android/CMakeLists.txt
+++ b/node_modules/react-native-worklets/android/CMakeLists.txt
@@ -94,7 +94,8 @@
 # build shared lib
 set_target_properties(worklets PROPERTIES LINKER_LANGUAGE CXX)
 
-target_link_libraries(worklets log ReactAndroid::jsi fbjni::fbjni)
+find_library(c++_shared_LIBRARY c++_shared)
+target_link_libraries(worklets log ReactAndroid::jsi fbjni::fbjni ${c++_shared_LIBRARY})
 
 if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
   target_link_libraries(worklets ReactAndroid::reactnative)
